src/map/map.ts

```diff
@@ -1119,4 +1239,72 @@ export class PropertiesMap {
         }
         return check;
     }
+
+    // this applies the opacity filter to the points, saving the points
+    // in the "main" trace (i.e. those meeting the filter criterion) in
+    // this._mainPoints, and putting all other points in this._backgroundPoints
+    private _updateFilter() {
+        const opacityMode = this._options.opacity.mode.value;
+
+        // don't know if there's a better way to do this
+        const nIdx = this._property(this._options.opacity.filter.property.value).values.length;
+        const allIdx = [];
+        for (let i = 0; i < nIdx; i++) {
+            allIdx.push(i);
+        }
+
+        let mP: Array<number> = allIdx;
+        let bP: Array<number> = [];
+
+        if (opacityMode === 'filter') {
+            const values = this._property(this._options.opacity.filter.property.value).values;
+            const operator = this._options.opacity.filter.operator.value;
+            const cutoff = this._options.opacity.filter.cutoff.value;
+
+            if (operator === '>') {
+                mP = allIdx.filter((i) => values[i] > cutoff);
+                bP = allIdx.filter((i) => values[i] <= cutoff);
+            } else if (operator === '<') {
+                mP = allIdx.filter((i) => values[i] < cutoff);
+                bP = allIdx.filter((i) => values[i] >= cutoff);
+            } else if (operator === '=') {
+                mP = allIdx.filter((i) => values[i] === cutoff);
+                bP = allIdx.filter((i) => values[i] !== cutoff);
+            } else {
+                throw Error('No other filters are supported');
+            }
+        }
+
+        this._mainPoints = mP;
+        this._backgroundPoints = bP;
+    }
+
+    // this is a *new* version of _selectTrace which not only returns the objects
+    // for the specific trace, but avoids unnecessary computation
+    private _filter<T>(objectsToFilter: T[], trace?: number, selectedObjects?: T[]): T[] | T[][] {
+        let mainObjects: T[] = objectsToFilter;
```

Comment:
naming: `mainObjects` => `main`, `objectsToFilter` => `all`, `backgroundObjects` => `background`

Reply:
all feels too vague for me. I wanted to be specific to say objects --> mainObjects, etc